data_provenance/hf_downloader.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import gzip
+import os
+import shutil
+
+from huggingface_hub import snapshot_download
+
+from licensed_pile.logs import configure_logging, get_logger
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Data Provenance Hugging Face Data Downloader"
+    )
+    parser.add_argument(
+        "--hf_dataset",
+        default="DataProvenanceInitiative/common_pile_ultra_permissive",
+        help="The label for the HuggingFace dataset that can be used in HuggingFace's load_dataset()",
+    )
+    parser.add_argument(
+        "--hf_token",
+        default=None,
+        help="Your API token for HuggingFace datasets",
+    )
+    parser.add_argument(
+        "--top_dir", default="data/raw-data-provenance", help="Path to output directory"
+    )
+    return parser.parse_args()
+
+
+def main(args):
+    logger = get_logger()
+    logger.info(f"Filtering to just the datasets in {args.hf_dataset}")
+
+    os.makedirs(args.top_dir, exist_ok=True)
+
+    logger.info(f"Downloading {args.hf_dataset}")
+
+    snapshot_download(
+        repo_id=args.hf_dataset,
+        repo_type="dataset",
+        local_dir=args.top_dir,
+        allow_patterns="*.jsonl",
+        token=args.hf_token,
+    )
+
+    logger.info(f"Saving {args.hf_dataset}")
+
+    hugging_dir = os.path.join(args.top_dir, ".huggingface")
+    if os.path.exists(hugging_dir):
+        shutil.rmtree(hugging_dir)
```

Comment:
I guess if we are going to be nuking this path we should be careful here, right? I.e. if hugging_dir exists when the script is called, we should exit to avoid nuking an existing huggingface dir?

Reply:
This is currently there to get around the issue when a previously cached dataset exists in that path then it will be nuked otherwise will throw an error from datasets. 

Can think about adding some edge cases there.

Reply:
My point is that if a user does something like specify ~ for --top_dir it would nuke their standard hugging face cache directory, which would be a pretty surprising result. I was just suggesting that you check whether --top_dir/.huggingface exists before putting stuff in it and then nuking it, because presumably --top_dir is supposed to be a dir where there's no .huggingface directory.

Reply:
Sounds good. Will add something for handling those cases.

Reply:
Something like
```python
hugging_dir = os.path.join(args.top_dir, ".huggingface")
if os.path.exists(hugging_dir):
    logger.error(f"The specified directory '{args.top_dir}' already contains a .huggingface folder. "
                 "Choose a different directory to avoid potential data loss.")
    return

os.makedirs(args.top_dir, exist_ok=True)
```

Reply:
and
```python
if top_dir == os.path.expanduser("~/.cache/huggingface"):
    logger.error(f"The specified directory '{top_dir}' is the default Hugging Face cache directory. "
                 "Choose a different directory to avoid data loss.")
    return
    ```

Reply:
```python
assert args.top_dir != os.path.expanduser("~/.cache/huggingface"), \
    f"The specified directory '{args.top_dir}' is the default Hugging Face cache directory. " \
    "Choose a different directory to avoid data loss."

hugging_dir = os.path.join(args.top_dir, ".huggingface")
assert not os.path.exists(hugging_dir), \
    f"The specified directory '{args.top_dir}' already contains a .huggingface folder. " \
    "Choose a different directory to avoid potential data loss."
    ```

Reply:
Add some asserts to prevent hf cache removal